xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceAccountService.java

```diff
@@ -91,22 +92,36 @@ public Object getRawAccountInfo() throws IOException {
   public Map<CurrencyPair, Fee> getDynamicTradingFees() throws IOException {
     try {
       BinanceAccountInformation acc = account();
-      BigDecimal makerFee =
-          acc.makerCommission.divide(new BigDecimal("10000"), 4, RoundingMode.UNNECESSARY);
-      BigDecimal takerFee =
-          acc.takerCommission.divide(new BigDecimal("10000"), 4, RoundingMode.UNNECESSARY);
-
       Map<CurrencyPair, Fee> tradingFees = new HashMap<>();
       List<CurrencyPair> pairs = exchange.getExchangeSymbols();
 
-      pairs.forEach(pair -> tradingFees.put(pair, new Fee(makerFee, takerFee)));
+      pairs.forEach(pair -> tradingFees.put(pair, getTradingFee(acc)));
 
       return tradingFees;
     } catch (BinanceException e) {
       throw BinanceErrorAdapter.adapt(e);
     }
   }
 
+  @Override
+  public Map<Instrument, Fee> getDynamicTradingFees(Set<Instrument> instruments) throws IOException {
+    try {
+      BinanceAccountInformation acc = account();
+      Map<Instrument, Fee> tradingFees = new HashMap<>();
+      instruments.forEach(instrument -> tradingFees.put(instrument, getTradingFee(acc)));
```

Comment:
Either 1) exclude instruments that are missing in exchange.getExchangeSymbols() or 2) throw exception when invalid instrument is queried

Reply:
done